app/views/conversations/_conversation.html.erb

```diff
@@ -1,24 +1,30 @@
-<li class="relative bg-white group hover:bg-gray-50 focus-within:ring-2 focus-within:ring-inset focus-within:ring-blue-600">
-  <%= tag.div class: "border-l-4 py-5 px-6 #{conversation.unread_messages_for?(current_user) ? "border-blue-400" : "border-white group-hover:border-gray-50"}" do %>
-    <div class="flex justify-between space-x-3">
-      <div class="min-w-0 flex-1">
-        <%= link_to conversation_path(conversation), class: "block focus:outline-none" do %>
-          <span class="absolute inset-0" aria-hidden="true"></span>
-          <h2 class="text-sm font-medium text-gray-900 truncate">
-            <% if conversation.deleted_business_or_developer? %>
-              <%= t("deleted_account") %>
-            <% else %>
-              <%= conversation.other_recipient(current_user).name %>
-            <% end %>
-          </h2>
-        <% end %>
+<div id="<%= dom_id conversation %>">
+  <li class="relative bg-white group hover:bg-gray-50 focus-within:ring-2 focus-within:ring-inset focus-within:ring-blue-600">
+    <%= tag.div class: "border-l-4 py-5 px-6 #{conversation.unread_messages_for?(current_user) ? "border-blue-400" : "border-white group-hover:border-gray-50"}" do %>
+      <div class="flex justify-between space-x-3">
+        <div class="min-w-0 flex-1">
+          <%= link_to conversation_path(conversation), class: "block focus:outline-none" do %>
+            <span class="absolute inset-0" aria-hidden="true"></span>
+            <h2 class="text-sm font-medium text-gray-900 truncate">
+              <% if conversation.deleted_business_or_developer? %>
+                <%= t("deleted_account") %>
+              <% else %>
+                <%= conversation.other_recipient(current_user).name %>
+              <% end %>
+            </h2>
+          <% end %>
+        </div>
+        <span class="shrink-0 whitespace-nowrap text-sm text-gray-500">
+          <%= render TimeComponent.new(conversation.messages.last.updated_at) %>
+        </span>
       </div>
-      <span class="shrink-0 whitespace-nowrap text-sm text-gray-500">
-        <%= render TimeComponent.new(conversation.messages.last.updated_at) %>
-      </span>
-    </div>
-    <div class="mt-1">
-      <p class="line-clamp-2 text-sm text-gray-600"><%= conversation.messages.last.body %></p>
-    </div>
-  <% end %>
-</li>
+      <div class="mt-1">
+        <p class="line-clamp-2 text-sm text-gray-600"><%= conversation.messages.last.body %></p>
+      </div>
+    <% end %>
+  </li>
+  <div class="text-right pt-2.5">
+   <%= button_to current_page?(conversations_path) ? "Archive" : "Unarchive",
```

Comment:
Please move these strings to the locale file.

Reply:
Means? Do you mean to shift the logic of the button to `index` files of their respective controllers? If yes, then how would we have access to the `id` for each of the `@conversations`?

Reply:
No, move “Archive” and “Unarchive” to `en.locale.yml`.

Reply:
Sorry! But I am not able to follow you. Could you elaborate on what exactly you want me to do?

Reply:
Plain text shouldn't exist in views, it should be moved to a translation file so different languages can use their own translation of the string.

https://guides.rubyonrails.org/i18n.html

So for this, you would add the following to `config/locales.en.yml`:

```yml
conversations:
  conversations:
    archive: Archive
    unarchive: Unarchive
```

Then in the view, line 27 becomes:

```erb
<%= button_to current_page?(conversations_path) ? t(".archive") : t(".unarchive") # ... %>
```

Does that help?

Reply:
Acha you meant the text! I thought you were talking about the logic of the `archive` & `unarchive` buttons and when you said in the `locales` I was confused because that's used for translation purposes! Thanks a lot for the detailed explanation! 

Done ✅ 😄